### PR TITLE
Story 12292 sms messaging api documentation

### PR DESCRIPTION
**Pull request by cgaroutte**
[Description of the change] and link to JIRA ticket if applicable.

https://invoca.atlassian.net/browse/STORY-12292

The rendered output of these changes is available for viewing here: https://developers.invoca.net/en/story-12292_sms_messaging_api_documentation/api_documentation/sms_messaging_api/index.html

This pull request implements the swagger-ui in the developer docs repo. The following files were copied directly from [https://github.com/swagger-api/swagger-ui/tree/master/dist](url)

- index.css
- swagger-ui.css
- swagger-ui.css.map
- swagger-initializer.js
- swagger-ui-bundle.js 
- swagger-ui-bundle.js.map

A new template file was added `_swagger_ui.txt`. The template is used in `source/api_documentation/sms_messaging_api/index.rst`. It takes a parameter `:id`. The template will use this parameter and create a div with the passed id (`swagger-ui` in this case). `swagger-initializer.js` looks for a div with id `swagger-ui` and if it finds it, replaces the div with the swagger ui.

The swagger ui css was overridden with customer rules in `source/_static/css/swagger-ui/index.css` in order to better match the current RTD styling. These will need to be more finely tuned in the future.

## Checklist

- [✔️ ] Find the [Service owning team](https://docs.google.com/spreadsheets/d/1YF2wuepY5SZTpVEdT9gwhTPhpnDGZeVjywEwvbJS5TQ/edit#gid=0&fvid=1520238175) for these changes, and tag the team as "Reviewers" on this PR
- [✔️ ] Test the documentation changes on readthedocs as a private branch
- [✔️ ] If changing general content, have agreement on whether to apply to latest version or all versions (if all versions, provide links to the related PRs below)

### CODE REVIEW

File: source/api_documentation/index.rst [hunk 1]
```diff
@@ -29,6 +29,9 @@ used to ingest calls from external sources for conversation intelligence process
 :doc:`calls_in_progress_api/index` -
 used to query/update calls that are currently in progress within the Invoca platform
 
+:doc:`sms_messaging_api/index` -
+used to enable/disable promo numbers for SMS, as well as send SMS messages
```
Comment (gabezurita) [inline source/api_documentation/index.rst]: Given the [MVP doc product features description](https://invoca.atlassian.net/wiki/spaces/DEV/pages/59409170433/SMS+MVP+Source+of+Truth+Doc#Project-Milestones-And-Product-Expectations) and maybe the [SMS alpha epic description](https://invoca.atlassian.net/browse/STORY-11414), how about we make this a bit more broad? Maybe:

`used for SMS number management, as well as handling incoming/outgoing SMS messages within the Invoca platform`

Would be good to get feedback from @jbuehring

File: source/_static/css/swagger-ui/index.css [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+.swagger-ui tr:nth-child(even) {
+  color: #636060 !important;
+  background-color: #FFF !important;
+}
+
+.swagger-ui tr:nth-child(odd) {
+  color: #4A4A4A !important;
+  background-color: #FFF !important;
+}
+*/
```
Comment (gabezurita) [inline source/_static/css/swagger-ui/index.css]: Did we want to just remove this?

File: source/api_documentation/sms_messaging_api/index.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+------------------------------------
+Filtering
+------------------------------------
+The records returned by `GET` requests can be filtered using URL parameters.
+Filters are comprised of:
+
+- a valid filterable field
+- a bracketed operator appended to the filterable field
+- a filter value separated from the filterable field and its associated operator by an equal sign (`=`)
```
Comment (gabezurita) [inline source/api_documentation/sms_messaging_api/index.rst:93]: @jbuehring, per [this thread's discussion](https://invoca.slack.com/archives/C042LLXBSHF/p1666908097415999?thread_ts=1666905760.149919&cid=C042LLXBSHF) (and pending further discussion), we may want to remove the above.

File: source/api_documentation/sms_messaging_api/index.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  * - empty
+    - Return only those records where the filtered field has an empty value such as integer 0, MySQL NULL/null, or an empty string
+    - deleted_at[empty]
+
+
+Each `GET` request denotes the valid filterable fields in its response body. Please see the API examples below for more information.
+
+If a filterable field is defined with a value but does not otherwise include a bracketed operator,
+it is assumed that the `[eq]` operator was intended.
+For example, `direction=incoming` means the same as `direction[eq]=incoming`.
```
Comment (gabezurita) [inline source/api_documentation/sms_messaging_api/index.rst]: Same as another comment in this PR, @jbuehring, per [this thread's discussion](https://invoca.slack.com/archives/C042LLXBSHF/p1666908097415999?thread_ts=1666905760.149919&cid=C042LLXBSHF) (and pending further discussion), we may want to remove the above.

**Review by gabezurita — COMMENTED** [review]
Awesome job Chris! I left a few questions/comments

Would be good to get the feedback from someone who has experience with the tools used in this PR, maybe someone outside the team?

File: source/api_documentation/index.rst (hunk 1 above)
Comment (cgaroutte) [reply source/api_documentation/index.rst]: Yeah, I like that. 👍

File: source/_static/css/swagger-ui/index.css (hunk 2 above)
Comment (cgaroutte) [reply source/_static/css/swagger-ui/index.css]: I believe so. I think I had kept it around while I was experimenting with simplifying the rules that are now commented out.

File: source/api_documentation/sms_messaging_api/index.rst (hunk 3 above)
Comment (cgaroutte) [reply source/api_documentation/sms_messaging_api/index.rst:93]: Going to modify to indicate the operator is required, per the changes in WEB-6455.

File: source/api_documentation/sms_messaging_api/index.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+Filtering
+------------------------------------
+The records returned by `GET` requests can be filtered using URL parameters.
+Filters are comprised of the following required components:
+
+- a valid filterable field
+- a bracketed operator appended to the filterable field
+- a filter value separated from the filterable field and its associated operator by an equal sign (`=`)
+
+For example: `https://example.com?created_at[lte]=2022-01-31` returns all records with a `created_at` value less than or equal to 2022-01-31.
```
Comment (gfaza) [inline source/api_documentation/sms_messaging_api/index.rst]: i think we need to go through and make sure all instances of dates get changed to datetimes in the correct format.

Comment (cgaroutte) [reply source/api_documentation/sms_messaging_api/index.rst]: I'll update to use datetimes, and add a section that indicates you can use the shorter form.

Comment (gfaza) [reply source/api_documentation/sms_messaging_api/index.rst]: i think we're avoiding support of the shorter form for now/alpha.

Comment (cgaroutte) [reply source/api_documentation/sms_messaging_api/index.rst]: I see. I was not aware of that. I'll change to just use full datetimes then.

File: source/api_documentation/index.rst (hunk 1 above)
Comment (jbuehring) [reply source/api_documentation/index.rst]: :shipit:

File: source/api_documentation/sms_messaging_api/index.rst (hunk 3 above)
Comment (jbuehring) [reply source/api_documentation/sms_messaging_api/index.rst:93]: 👑

File: source/api_documentation/sms_messaging_api/index.rst (hunk 4 above)
Comment (jbuehring) [reply source/api_documentation/sms_messaging_api/index.rst]: 👑

File: source/api_documentation/sms_messaging_api/index.rst [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+Overview
+------------------------------------
+The SMS Messaging API provides the following functionality:
+
+- manage SMS-enabled phone numbers
+- send SMS messages
+- retrieve SMS messages and their associated Delivery Reports
+
+------------------------------------
+Authentication and Authorization
```
Comment (jbuehring) [inline source/api_documentation/sms_messaging_api/index.rst:22]: I would call out here that, unlike _other_ Invoca API, SMS Messaging will _only_ support an `Authorization` "Bearer" header, no use of `GET` or `POST` parameter named `oauth_token` is supported.

Comment (cgaroutte) [reply source/api_documentation/sms_messaging_api/index.rst:22]: That's a good point. 👍

Comment (cgaroutte) [reply source/api_documentation/sms_messaging_api/index.rst:22]: Updated.

**Review by jbuehring — APPROVED** [review]
Looks like it hits the right notes, especially esoteric things like how filters combine to form a chain of `AND` only.

File: source/api_documentation/sms_messaging_api/index.rst (hunk 5 above)
Comment (jbuehring) [reply source/api_documentation/sms_messaging_api/index.rst]: 👑

**Comment by gfaza** [comment]
hello @jpterry!  adding you for some visibility to this just-about-to-get-merged PR.  it's primarily composed of:
* CSS and layout customization
* inline support content pertaining to the SMS API
* a python script, that automates the transformation of `rspec-openapi` output yaml, to our developer-docs HTML, by way of swagger-ui.

**_update:_**
I've just merged some stuffs in, making this "The" PR.  sorry for the thrash.

**Pull request merged**